Army-Java/Units/Soldier.java

```diff
@@ -0,0 +1,14 @@
+package com.gymfox.Army.Units;
+
+import com.gymfox.Army.Ability.SoldierAbility;
+import com.gymfox.Army.State.HumanState;
+
+public class Soldier extends Unit {
+    public Soldier(String name, int healthPointLimit, int damage) {
+        super(name, healthPointLimit, damage);
+        setAbility(new SoldierAbility(this));
+        setUnitType(UnitType.SOLDIER);
+        setCurrentState(new HumanState(this));
+        setNextState(new HumanState(this));
```

Comment:
Looks not good. You should pass all these arguments to parent constructor as all these fields belong to the parent. So it looks a bit odd when you're setting parent's fields in its children.
Not only in this class but in all other `Unit` subclasses as well